provisioning/transport/mqtt/src/mqtt_ws.ts

```diff
@@ -0,0 +1,30 @@
+
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+'use strict';
+
+import { Mqtt } from './mqtt';
+import { RegistrationRequest } from 'azure-iot-provisioning-device';
+import { MqttBase } from 'azure-iot-mqtt-base';
+
+export class MqttWs extends Mqtt {
+
+  constructor(mqttBase?: MqttBase) {
+    super(mqttBase);
+  }
+
+    /* Codes_SRS_NODE_PROVISIONING_MQTT_18_049: [ When connecting using websockets, `Mqtt`Ws shall set the uri passed into the transport to 'wss://<host>:443/$iothub/websocket'.] */
+    modifyRequest(request: RegistrationRequest): RegistrationRequest {
```

Comment:
>modifyRequest [](start = 4, length = 13)

in another transport in the device client, both classes (would be Mqtt and MqttWs) have a _getConnectionUri() method that returns different things (in that case it'd be wss://... or mqtts://... . Then you just call _getConnectionUri in MQTT class. that way you don't have to override registrationrequest and queryOperationStatus in MqttWs,  and you don't have to test if the URI start with WSS anywhere.